crates/rmf_site_editor/src/site/physical_camera.rs

```diff
@@ -59,3 +61,125 @@ pub fn add_physical_camera_visuals(
         commands.entity(e).add_children(&[child]);
     }
 }
+
+pub fn check_double_click_event(
+    mut select: EventReader<Select>,
+    selection: ResMut<Selection>,
+    mut double_clicked: ResMut<DoubleClickSelection>,
+) {
+    let current_time = Instant::now();
+
+    for _ in select.read() {
+        if let (Some(last_entity), Some(last_time)) = (
+            double_clicked.last_selected_entity,
+            double_clicked.last_selected_time,
+        ) {
+            let elapsed_time = current_time.duration_since(last_time).as_millis();
+
+            if let Some(selected_entity) = selection.0 {
+                if last_entity == selected_entity && elapsed_time < 500 {
+                    double_clicked.entity = selection.0;
+                }
+            }
+        }
+        double_clicked.last_selected_entity = selection.0;
+        double_clicked.last_selected_time = Some(current_time);
+    }
+}
+
+pub fn focus_camera_on_double_clicked_object(
+    time: Res<Time>,
+    active_camera: ActiveCameraQuery,
+    mut transforms: Query<(&mut GlobalTransform, &mut Transform)>,
+    lanes: Query<&LaneSegments>,
+    walls: Query<&Edge<Entity>, With<WallMarker>>,
+    floors: Query<&Path<Entity>, With<FloorMarker>>,
+    measurements: Query<&Edge<Entity>, With<MeasurementMarker>>,
+    mut double_clicking: ResMut<DoubleClickSelection>,
+) {
+    if double_clicking.entity.is_none() {
+        return;
+    }
+
+    let mut target_position;
+    if let Some(double_clicked) = double_clicking.entity {
```

Comment:
The returning if the entity is none, plus the if let below (with an else branch that returns) are effectively the same thing. I would suggest refactoring every

```
if let Some([...]) = [...] {
} else { 
   [...]
   return; 
}
```

Into a

```
let Some([...]) = [...] else {
   [...]
   return;
};
```
This will reduce indentation and make sure we don't forget an else statement